### PR TITLE
API: Deprecate max_ori_out

### LINKED ISSUE #9967: Why is `max_ori_out` in `_apply_lcmv` enforced to be "signed"?
Dear mne developers, 

thanks for your great work. 

When I looked through the `beamformer` code in order to find out what the options for the parameter `max_ori_out` in `apply_lcmv` are, I found that this parameter is actually enforced to be "signed" with no alternative, while later in the code the option exists for it to be "abs". 

First lines of the function:
https://github.com/mne-tools/mne-python/blob/c2aaaa8aad83d62ee36b7030bca56d50315020e4/mne/beamformer/_lcmv.py#L203-L207

Later:
https://github.com/mne-tools/mne-python/blob/c2aaaa8aad83d62ee36b7030bca56d50315020e4/mne/beamformer/_lcmv.py#L242-L243

It is obvious that one of these two code snippets do not work as intended. What is the intended behaviour?

Kind regards,
Daniel

### PR DESCRIPTION
**Pull request by larsoner**
Closes #9967

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

File: mne/beamformer/_lcmv.py [hunk 1]
```diff
@@ -255,8 +249,14 @@ def _apply_lcmv(data, filters, info, tmin, max_ori_out):
...
 
+def _deprecate_max_ori_out(max_ori_out):
+    if max_ori_out is not None:
+        warn('max_ori_out will be removed in 1.0, do not pass it as an '
+             'argument', DeprecationWarning)
+
+
 @verbose
-def apply_lcmv(evoked, filters, max_ori_out='signed', verbose=None):
+def apply_lcmv(evoked, filters, *, max_ori_out=None, verbose=None):
```
Comment (britta-wstnr) [inline mne/beamformer/_lcmv.py:259]: I am probably being blind, but: why do we need the * operator here?

**Review by britta-wstnr — COMMENTED** [review]
Just a question, otherwise LGTM!

File: mne/beamformer/_lcmv.py (hunk 1 above)
Comment (larsoner) [reply mne/beamformer/_lcmv.py:259]: It forces it to be input as a keyword argument only. In other words, this is not allowed:
```
apply_lcmv(evoked, filters, 'signed')
```

Comment (britta-wstnr) [reply mne/beamformer/_lcmv.py:259]: Aaah, that makes sense of course. Thanks for explaining!

**Review by britta-wstnr — APPROVED** [review]

**Comment by britta-wstnr** [comment]
Sorry, I mis-clicked - I approve :)

**Pull request merged**